test/transfer-manager.ts

```diff
@@ -443,12 +348,69 @@ describe('Transfer Manager', () => {
         fakeHelper.initiateUpload.rejects(new Error(expectedErr.message));
         fakeHelper.uploadPart.resolves();
         fakeHelper.completeUpload.resolves();
+        fakeHelper.abortUpload.resolves();
         return fakeHelper;
       };
       assert.rejects(
         transferManager.uploadFileInChunks(path, {}, mockGeneratorFunction),
         expectedErr
       );
     });
+
+    it('should pass through headers to initiateUpload', async () => {
+      const headersToAdd = {
+        'Content-Type': 'foo/bar',
+        'x-goog-meta-foo': 'foobar',
+      };
+
+      mockGeneratorFunction = (bucket, fileName, uploadId, partsMap) => {
+        fakeHelper = sandbox.createStubInstance(FakeXMLHelper);
+        fakeHelper.uploadId = uploadId || '';
+        fakeHelper.partsMap = partsMap || new Map<number, string>();
+        fakeHelper.initiateUpload.callsFake(headers => {
+          assert.deepStrictEqual(headers, headersToAdd);
+          return Promise.resolve();
+        });
+        fakeHelper.uploadPart.resolves();
+        fakeHelper.completeUpload.resolves();
+        fakeHelper.abortUpload.resolves();
+        return fakeHelper;
+      };
+
+      await transferManager.uploadFileInChunks(
+        path,
+        {headers: headersToAdd},
+        mockGeneratorFunction
+      );
+    });
+
+    it('should call abortUpload when passed the option and uploadID', async () => {
```

Comment:
This seems like a hack? Why not just expose the abort API and ask for an id? We could document that users should catch failures and call this method (since we're not doing it for them)/

Reply:
When we implemented this we decided not to expose the actual XML implementation to avoid users accessing it directly. This was done because at the time we did not know if other MPU implementations would come along and the default would be changed (i.e. JSON).

Reply:
Yeah it feels odd to have this API do two completely different things based on the parameters.
Instead, I would consider following python and [catching any errors thrown](https://github.com/googleapis/python-storage/pull/1115/files/43fd95c91fdafe61eac65e46a471684586e5a615#diff-39908d1c6ff23e82acdf23e96d17f7237b47a41b53d4de55adddca92b3fa6df2R1045) by MPU with a note:

```
The library will attempt to cancel uploads that fail due to an exception.
    If the upload fails in a way that precludes cancellation, such as a
    hardware failure, process termination, or power outage, then the incomplete
    upload may persist indefinitely. To mitigate this, set the
    `AbortIncompleteMultipartUpload` with a nonzero `Age` in bucket lifecycle
    rules, or refer to the XML API documentation linked above to learn more
    about how to list and delete individual downloads.
    ```

Reply:
This goes back to the debate of autodelete vs not. If we want to proceed with autodelete, I'll just move the logic internally to call the delete function.

Reply:
@frankyn ?

Reply:
I set up a discussion with @andrewsg and @ddelgrosso1 to suss this out today; will report back;